corehq/messaging/smsbackends/amazon_pinpoint/models.py

```diff
@@ -0,0 +1,64 @@
+import boto3
+from corehq.apps.sms.models import SQLSMSBackend
+from corehq.apps.sms.util import clean_phone_number
+from corehq.messaging.smsbackends.amazon_pinpoint.forms import PinpointBackendForm
+
+MESSAGE_TYPE = "TRANSACTIONAL"
+
+
+class SQLPinpointBackend(SQLSMSBackend):
+
+    class Meta(object):
+        app_label = 'sms'
+        proxy = True
+
+    @classmethod
+    def get_available_extra_fields(cls):
+        return [
+            'reply_to_phone_number',
+            'project_id',
+            'region',
+            'access_key',
+            'secret_access_key'
+        ]
+
+    @classmethod
+    def get_api_id(cls):
+        return 'PINPOINT'
+
+    @classmethod
+    def get_generic_name(cls):
+        return "Amazon Pinpoint"
+
+    @classmethod
+    def get_form_class(cls):
+        return PinpointBackendForm
+
+    def send(self, msg, *args, **kwargs):
+        phone_number = clean_phone_number(msg.phone_number)
+        config = self.config
+        client = boto3.client(
+            'pinpoint',
+            region_name=config.region,
+            aws_access_key_id=config.access_key,
+            aws_secret_access_key=config.secret_access_key
+        )
+        message_request = {
+            'Addresses': {
+                phone_number: {
+                    'ChannelType': 'SMS'
+                }
+            },
+            'MessageConfiguration': {
+                'SMSMessage': {
+                    'Body': msg.text,
+                    'MessageType': MESSAGE_TYPE,
+                    'OriginationNumber': config.reply_to_phone_number
+                }
+            }
+        }
+        response = client.send_messages(
+            ApplicationId=config.project_id,
+            MessageRequest=message_request
+        )
+        return response
```

Comment:
`send()` need not return a value, but it should parse the response and set error details on the `msg` object if necessary. The [Trumpia backend](https://github.com/dimagi/commcare-hq/blob/ac7aa4c23fde62339b6506fc94bdf9f81e865971/corehq/messaging/smsbackends/trumpia/models.py) can be used as an example.

Reply:
Makes sense. Thanks it's fixed. While working on this, I realized that msg object is not getting updated after assigning msg.backend_message_id = data["requestID"] [here](https://github.com/dimagi/commcare-hq/blob/ac7aa4c23fde62339b6506fc94bdf9f81e865971/corehq/messaging/smsbackends/trumpia/models.py#L63).
So I've added msg.save() after that line.

Reply:
The [part of the SMS system that calls `backend.send(...)`](https://github.com/dimagi/commcare-hq/blob/ac7aa4c23fde62339b6506fc94bdf9f81e865971/corehq/apps/sms/api.py#L319) eventually calls `msg.save()`, so it should not be necessary to save in `send()` in most cases.